src/event-stream/reader.ts

```diff
@@ -66,12 +76,81 @@ export function getAddressFromPublicKeyHash(
   return addrString;
 }
 
-export function createTransactionFromCoreBtcTxEvent(event: StxTransferEvent): Transaction {
+export function createTransactionFromCoreBtcStxLockEvent(
+  chainId: ChainID,
+  event: StxLockEvent,
+  burnBlockHeight: number,
+  txResult: string
+): Transaction {
+  const resultCv = deserializeCV(Buffer.from(txResult.substr(2), 'hex')) as ResponseOkCV;
+  const resultTuple = resultCv.value as TupleCV;
+  const lockAmount = resultTuple.data['lock-amount'] as UIntCV;
+  const stacker = resultTuple.data['stacker'] as StandardPrincipalCV;
+  const unlockBurnHeight = (resultTuple.data['unlock-burn-height'] as UIntCV).value.toNumber();
+
+  // Number of cycles: floor((unlock-burn-height - burn-height) / reward-cycle-length)
+  const rewardCycleLength = chainId === ChainID.Mainnet ? 2100 : 50;
+  const lockPeriod = Math.floor((unlockBurnHeight - burnBlockHeight) / rewardCycleLength);
+  const senderAddress = createAddress(event.stx_lock_event.locked_address);
+  const poxAddress = createAddress(
+    chainId === ChainID.Mainnet ? 'SP000000000000000000002Q6VF78' : 'ST000000000000000000002AMW42H'
+  );
+
+  const clarityFnArgs: ClarityValue[] = [
+    lockAmount,
+    tupleCV({
```

Comment:
Can you open an issue to track mocking a more accurate argument here -- this address should be the first output address of the bitcoin transaction. I realize that the `stacks-node` doesn't currently provide enough information to obtain this, so changes would be required there as well, but we should discuss in a new issue.

Reply:
Created https://github.com/blockstack/stacks-blockchain-api/issues/415